new-lamassu-admin/src/pages/Blacklist/Blacklist.js

```diff
@@ -0,0 +1,183 @@
+import { fromNamespace, toNamespace } from 'src/utils/config'
+
+import styles from './Blacklist.styles'
+import BlackListModal from './BlacklistModal'
+import BlacklistTable from './BlacklistTable'
+
+const useStyles = makeStyles(styles)
+
+const groupByCode = R.groupBy(obj => obj.cryptoCode)
+
+const DELETE_ROW = gql`
+  mutation DeleteBlacklistRow($cryptoCode: String!, $address: String!) {
+    deleteBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const GET_BLACKLIST = gql`
+  query getBlacklistData {
+    blacklist {
+      cryptoCode
+      address
+    }
+    cryptoCurrencies {
+      display
+      code
+    }
+  }
+`
+
+const SAVE_CONFIG = gql`
+  mutation Save($config: JSONObject) {
+    saveConfig(config: $config)
+  }
+`
+
+const GET_INFO = gql`
+  query getData {
+    config
+  }
+`
+
+const ADD_ROW = gql`
+  mutation InsertBlacklistRow($cryptoCode: String!, $address: String!) {
+    insertBlacklistRow(cryptoCode: $cryptoCode, address: $address) {
+      cryptoCode
+      address
+    }
+  }
+`
+
+const Blacklist = () => {
+  const { data: blacklistResponse } = useQuery(GET_BLACKLIST)
+  const { data: configData } = useQuery(GET_INFO)
+  const [showModal, setShowModal] = useState(false)
+  const [clickedItem, setClickedItem] = useState({
+    code: 'BTC',
+    display: 'Bitcoin'
+  })
+  const [deleteEntry] = useMutation(DELETE_ROW, {
+    onError: () => console.error('Error while deleting row'),
+    refetchQueries: () => ['getBlacklistData']
+  })
+
+  const [addEntry] = useMutation(ADD_ROW, {
+    onError: () => console.error('Error while adding row'),
+    onCompleted: () => setShowModal(false),
+    refetchQueries: () => ['getBlacklistData']
+  })
+
+  const [saveConfig] = useMutation(SAVE_CONFIG, {
+    refetchQueries: () => ['getData']
+  })
+
+  const classes = useStyles()
+
+  const blacklistData = R.path(['blacklist'])(blacklistResponse) ?? []
+  const availableCurrencies =
+    R.path(['cryptoCurrencies'], blacklistResponse) ?? []
+
+  const formattedData = groupByCode(blacklistData)
+
+  const complianceConfig =
+    configData?.config && fromNamespace('compliance')(configData.config)
+
+  const rejectAddressReuse = complianceConfig?.rejectAddressReuse ?? false
+
+  const addressReuseSave = rawConfig => {
+    const config = toNamespace('compliance')(rawConfig)
+    return saveConfig({ variables: { config } })
+  }
+
+  const onClickSidebarItem = e => {
+    setClickedItem({ code: e.code, display: e.display })
+  }
+
+  const handleDeleteEntry = (cryptoCode, address) => {
+    deleteEntry({ variables: { cryptoCode, address } })
+  }
+
+  const toggleModal = () => setShowModal(!showModal)
+
+  const addToBlacklist = (cryptoCode, address) => {
+    addEntry({ variables: { cryptoCode, address } })
+  }
+
+  return (
+    <>
+      <TitleSection title="Blacklisted addresses">
+        <Link onClick={toggleModal}>Blacklist new addresses</Link>
```

Comment:
Just like we don't want toggle on closing the modal, we don't want it here. This button should only set the modal to true. 

`() => setShowModal(true)`